pyvista/core/pointset.py

```diff
@@ -1280,11 +1280,18 @@ def plot_curvature(self, curv_type='mean', **kwargs):
     def volume(self) -> float:
         """Compute the volume of the point grid.
 
-        This extracts the external surface and computes the interior
-        volume.
+        .. versionchanged:: 0.37.0
+           This now calculates the sum of the volume of all cells.
+
         """
-        surf = self.extract_surface().triangulate()
-        return surf.volume
+        vols = self.compute_cell_sizes(length=False, area=False, volume=True)
```

Comment:
If the inaccuracy comes from `vtkMassProperties`, do you think we could/should replace that for `PolyData` as well?

Reply:
Since PolyData cells never fill a volume, i.e
it is a 2D surface that sometimes encloses a volume, I think this is the only option (although I'm not much of a PolyData user).  `area` for `PolyData` is calculated using the `compute_cell_sizes` as in this PR.

Reply:
For a moment I got confused between `triangulate()` and Delaunay triangulation! We do the former but not the latter. If we did Delaunay then we could compute exact sizes. But in the general case if we only know the surface is closed, we can't easily compute a 3d triangulation (because the surface could easily be concave). Considering concave surface meshes instantly explains to me why we can only use an approximation.

At best we could leave a note for `PolyData.volume` that this is an approximation, and even that would be outside this PR. I think we're good here, thanks.